src/json.rs

```diff
@@ -0,0 +1,136 @@
+use crate::{
+    extract::{has_content_type, rejection::*, take_body, FromRequest, RequestParts},
+    prelude::response::IntoResponse,
+    Body,
+};
+use async_trait::async_trait;
+use http::{
+    header::{self, HeaderValue},
+    StatusCode,
+};
+use hyper::Response;
+use serde::{de::DeserializeOwned, Serialize};
+use std::ops::Deref;
+
+/// JSON Extractor/Response
+///
+/// When used as an Extractor, it can deserializes request bodies into some type that
+/// implements [`serde::Serialize`]. If the request bodies cannot be parsed, or it does not contain
+/// the `Content-Type: application/json`t header, will reject the request and return a
+/// `400 Bad Request` response.
```

Comment:
```suggestion
/// JSON Extractor/Response
///
/// When used as an extractor, it can deserialize request bodies into some type that
/// implements [`serde::Serialize`]. If the request body cannot be parsed, or it does not contain
/// the `Content-Type: application/json` header, it will reject the request and return a
/// `400 Bad Request` response.
```